### ISSUE TITLE
Unwanted empty lines in reply/forward headers in plain-text emails

### ISSUE THREAD

**Issue by ckeydel** [RealRaven2000/SmartTemplate4#54]
Normally I use HTML emails and I have set my reply and forward headers the way I want them, with the "Use HTML" checkbox checked, the "Replace line breaks with \<br\>" unchecked (per the author's suggestion), and the \<br\> tags put directly in the quote header.
With some people, I want to use plain-text format in replies because they send such emails to me and have requested that I reply in kind. So as usual in TB, I keep the \<Shift\> key pressed while hitting the reply or forward button to compose the email in plain-text format. This works in principle but two issues:
- in the compose window there is a huge amount of white space before the signature, and also more white space than usual after the signature
- there are extra empty lines between "From:", "Sent:" etc.

The extra lines in the reply header come from the HTML header's \<br\> tags plus the regular line breaks. In HTML, they are interpreted correctly, i.e. don't lead to extra line breaks. In plain text, both are interpreted as line breaks. I guess that same thing happens with the HTML signature.

I can work around it by making the HTML reply header one super-long single line with \<br\> tags in between but it's not exactly nice. The regular line breaks should be simply ignored for plain-text emails if it is a "Use HTML" header.

BTW this doesn't happen if "Replace line breaks with \<br\>" is checked and I remove the \<br\> tags. But I don't want to use this option anymore because of other issues.

**Comment by RealRaven2000** [comment]
It would be helpful if you could export your account settings using the save template feature:

![image](https://user-images.githubusercontent.com/6314674/74520793-2fe79a80-4f10-11ea-8537-6ddb44426738.png)

and also your signature. You may send them to me directly off github to preserve privacy. But you can make a screenshot here of your Composition & Addressing settings in Tb accounts:

![image](https://user-images.githubusercontent.com/6314674/74520893-5c031b80-4f10-11ea-9e69-868cfac6a31c.png)

**Comment by ckeydel** [comment]
Just sent this to you using GitHub direct email.

**Comment by RealRaven2000** [comment]
hmm, not sure if I got that... can you send it to realraven2000@hotmail.com?

**Comment by ckeydel** [comment]
done.

**Comment by RealRaven2000** [comment]
thanks, I got it now. Turns out Hotmail now also have a aggressive anti Spam policy like google, so I to recover the mail to see it on my POP3 account. I understand your default is HTML (like most) and you force plaintext by holding SHIFT when clicking on the reply button.. 

Should  be easy enough to replicate. plaint text is the unloved stepchild of the SmartTemplate developer team (me). ;)

**Comment by RealRaven2000** [comment]
Ok, I managed to load your template, let's have a look at the output when I force text mode with SHIFT:

![image](https://user-images.githubusercontent.com/6314674/74591307-dade7e00-500e-11ea-9d94-c5e5afa0db46.png)

it looks like there is additional space on the quote header lines (From: , Sent: , Cc: etc.). In source code it looks quite innocuous:

```
<!DOCTYPE html><html>
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
  </head>
  <body style="font-family: -moz-fixed; white-space: pre-wrap; width: 72ch;" smarttemplateinserted="true">
      <div id="smartTemplate4-template"><p> </p></div>
      <div id="smartTemplate4-quoteHeader">
<hr>
<b>From:</b> Axel (Thunderbird Daily 67) &lt;thunderbirddaily67@gmail.com&gt;<br>
<b>Sent:</b>  Saturday, 15.02.2020 16:13 UTC<br>
<b>To:</b> Axel (Development) &lt;development@quickfolders.org&gt;<br>
<b>Cc:</b> <br>
<b>Subject:</b> [SmartTemplate4] test - issue 54 (unwanted empty lines in plain-text)<br>
</div>

<span style="white-space: pre-wrap; display: block; width: 98vw;">&gt; Dear Axel (Thunderbird Daily 67)<br>
```
Once I delete the extra line breaks in editor, this removers the <br> parts. Relevant code portion:

```
<hr>
<b>From:</b> Axel (Thunderbird Daily 67) &lt;thunderbirddaily67@gmail.com&gt;
<b>Sent:</b>  Saturday, 15.02.2020 16:13 UTC
<b>To:</b> Axel (Development) &lt;development@quickfolders.org&gt;
<b>Cc:</b> 
<b>Subject:</b> [SmartTemplate4] test - issue 54 (unwanted empty lines in plain-text)<br>
</div>
```

So this means the text node treats textual line breaks (and not just `<br>` tags) as line breaks (unlike HTML source code). I need to find out which part of the "mail-building" code inserts the <br> or the line breaks.

**Comment by RealRaven2000** [comment]
If you want to hear an amusing episode, I was just debugging SmartTempalte4 (this particular bug) while error checking a broken folder and listening to a YouTube video (talk about multi-tasking). Incidentally the SmartTemplate/Support folder was broken...
.... and then I got a BSOD
![image](https://user-images.githubusercontent.com/6314674/74613393-02fcd880-5106-11ea-99bc-2cac3c83cefa.png)
with a hard stop, not even being able to dump the memory. After hard restarting I had to find that my main HD (2 terabytes, divided into 2 equally sized partitions, D: for music/studio recording and E: for development) had decided to go to the happy hunting grounds in the sky. 

Unfortunately, I didn't do a complete backup, but most of the current Add-on projects were saved. My recording system was not so lucky so I proceeded to call off all studio projects for the next week... I am still committed to fixing this bug but it will be a little delayed as I need to dedicate a major part of my energy to "getting up and running again". 

Pretty happy that this particular issue (Unwanted empty lines, not the dat loss) should be easy to fix though. Onwards and upwards, I guess.

**Comment by RealRaven2000** [comment]
... having said this: I just reserved some space on another partition with some restored data, so here is a test version with an attempted fix:

[smartTemplate-fx-2.9.2pre5.zip](https://github.com/RealRaven2000/SmartTemplate4/files/4211333/smartTemplate-fx-2.9.2pre5.zip)

It only will fix the quote text if the following conditions apply
- the Editor is in text only mode 
- the quote contains <br> or <p> elements. I take this as an assumption that you are not interested in replacing line breaks. (should probably use the [x] replace line breaks option instead, but this is quick and dirty and works for your example) - I need to think this through more...

```
  if (!IsHTMLEditor()) {
    // [issue 54] extra line spaces in (html) quote header when replying text only.
    // if template contains <br> or <p> let's strip out "formatting" text content line breaks.
    if (quoteHeader.includes("<br>") || quoteHeader.includes("<p>"))
      quoteHeader =  quoteHeader.replace(/\n/gm,"")
  }
```

**Comment by ckeydel** [comment]
Sorry to hear about your hard drive incident! I still have some stickers somewhere to help prevent future such catastrophes. Wait ... where are they. Ah, yes, here:
![image](https://user-images.githubusercontent.com/52654513/74634822-fa87ba80-5164-11ea-831e-b73ad5afefdc.png)
Sorry, couldn't resist! ;-)
Seriously, though, glad you got up and running relatively fast.

**Comment by ckeydel** [comment]
I tried v2.9.2pre5 (still reported as v2.9.1). Two issues:
In my reply header, the last two lines are
`<b>Cc:</b> %cc%<br>
<b>Subject:</b> %subject%<br>`

**Comment by RealRaven2000** [comment]
> I tried v2.9.2pre5 (still reported as v2.9.1). Two issues:
> First, in my reply header, the last two lines are:
> 
> `<b>Cc:</b> %cc%<br>`
> `<b>Subject:</b> %subject%<br>`
> 
> but in the plain-text response email, there is no extra empty line between the reply header and the body text, which is odd:
> 
> `Cc: Some Name <some@email.de>`
> `Subject: Re: Whatever subject`
> `> Yes! Some body text.`

That looks like what you asked it to do.... blockquote starts straight after the last (single) <br> . Why should SmartTemplate insert additional brs in plaintext?
> 
> Since there will never be a case where the first line of body text is appended to the subject line

You mean quoted text, right? Blockquote is a display:block level element and as such it always is in a separate line. So thinking about this in html layout there should probably be an empty line visible. Try adding a &nbsp; after the last <br>

>, the last <br> in the reply header should mean that there is one empty line. This would also be consistent with the HTML handling where there is also some space between the last line of the header and the first line of body text.
> 
> Secondly, the changes did nothing for the large amount of extra white space around the signature, "where you write your response text".

Your template made no sense to me- it's only %sig(2)% and noting else - no salutation, no cursor - why? Where do you write, above or below your signature?

Also i don't believe i have your signature, is it HTML  or plain tex.?

**Comment by ckeydel** [comment]
Yes, adding a second \<br\> after %subject% with create the right amount of spacing in both cases - plain-text and HTML, so this part can be considered closed IMO.

My default signature I sent you in my email on 2/14, both as screenshot and as HTML text. There is nothing special in it but it is in HTML format and therefore difficult to post here because the tags are getting removed. %cursor% is part of that signature not the template, I believe you sent me that whole signature as a suggestion in the past and I just copied it. I am writing above the signature - again, this info was in my email. I am using the same composition settings as you do.

Yes, I only use %sig(2)% here and add anything else myself on a case-by-case basis. Shouldn't be a problem, should it?

**Comment by RealRaven2000** [comment]
>  but in the plain-text response email, there is no extra empty line between the reply header and the body text, which is odd:

Cc: Some Name <some@email.de>
Subject: Re: Whatever subject
> Yes! Some body text.

(now on my desktop browser) this looks correct to me. This is the default behavior, with SmartTemplate disabled:
![image](https://user-images.githubusercontent.com/6314674/74650122-e8177c00-5178-11ea-91f3-abd644c18af3.png)

so as you can see - no extra space between the quote header "_`On 15/02/2020 16:13, Thunderbirddaily 67 wrote:`_" and the quoted text "_`> Dear Axel (Thunderbird Daily 67)...`_"

So the new line removal seems to work fine. and SmartTemplate is replacing the quote header line with whatever you give it. In source code (plain text) that Thunderbird standard quote header followed by blockquote part looks like this:

`<div class="moz-cite-prefix">On 15/02/2020 16:13, Thunderbirddaily 67 wrote:<br></div>`

In source, there is an extra `<br>` put there by Thunderbird but id doesn't show as an empty line as it is collapsed within the line break of the closing `<div>`. To uncollapse, you can force the empty line to show up by adding a non-breaking space after the `<br>` like this:

`<div class="moz-cite-prefix">On 15/02/2020 16:13, Thunderbirddaily 67 wrote:<br>&nbsp;</div>`

which will give you this result:

![image](https://user-images.githubusercontent.com/6314674/74650604-fc0fad80-5179-11ea-8534-968a71be3f95.png)

**Comment by ckeydel** [comment]
Yes, you're right, the &nbsp; works, too. Something new learned today... :-)
So the reply header handling for plain-text emails is "good" now I think.

**Comment by RealRaven2000** [comment]
> 
> 
> Yes, adding a second <br> after %subject% will create the right amount of spacing in both cases - plain-text and HTML, so this part can be considered closed IMO.
> 
> My default signature I sent you in my email on 2/14, both as screenshot and as HTML text. There is nothing special in it but it is in HTML format and therefore difficult to post here because the tags are getting removed. 

alright - in the future just use highlight any code and then use the `<>` button to wrap it so it doesn't actually get removed. It's hard to do on a mobile phone (which I replied to with earlier, so I had to edit on my desktop browser).

> %cursor% is part of that signature not the template, I believe you sent me that whole signature as a suggestion in the past and I just copied it. 

I think that's a little bit unfortunate, I would instead take it out of the  signature and would  instead change the template to something like this:

```
%cursor% <br>
%sig(2)%
```

> I am writing above the signature - again, this info was in my email. 

Yeah I didn't dig out the email, I get soo much support mail that it is hard to keep track :)

> Yes, I only use %sig(2)% here and add anything else myself on a case-by-case basis. Shouldn't be a problem, should it?

Not a problem, just semantically what you are doing is "writing in the signature" and not in the template, so it may present some problems if you try to style in HTML later. Personally my default (and recommended) pattern is like this:

```
<style> ... (all my CSS rules for my mail body) </style>
<div id="mytemplate">
  <p> (automated salutation) </p>
  <p> %cursor% </p>
  <p> yours sincerely, </p>
  %sig(html)%
</div>
```

I can use that as a common template with different signatures - you obviously do not need the following optional parts:

- styling (if you trust Thunderbird to do that for you and don't care what it looks like or want the defaults of the mail client person you send to decide)
- first line (salutation) - I usually have %from(name)% in a reply because I am lazy. I think you prefer to type it manually.
- third line (the goodbye line) - again, me being lazy
- `<p>` for writing all my email text within Paragraphs, I like that because it provides more natural spacing than mashing the new line twice every time, of course you can do this  "type writer style" editing too, but I always tend to think my style rules should determine the space between paragraphs. Also I dislike using `<br><br><br>` for "layout".
And I am lazy :)

so at its core you are left with this minimal approach:

```
 <div id="ck">
   %cursor%<br>
   %sig(html)%
 </div>
```

and then you could do a bit of trickery for styling stuff together and  you can just start typinh above the signature like you used to. I am actually planning to implement a new `%style(path)%` command for the future so I can share a style sheet between different templates for the same product.

**Comment by ckeydel** [comment]
That's nice! I started playing with it and it really is better.
Only the indentation, that doesn't play nice with plain-text emails because they are put before the cursor and the signature as real spaces.

**Comment by ckeydel** [comment]
Nah, for the signature the spacing is still too big. I have used your core/minimal approach for the ST4 signature, just removing the identation and I have defined a really simple signature in TB for testing:
`<p>Best Regards,<br>`
`Chris`
`</p>`
So the "Chris" should be in the next line after "Best Regards", right? This is indeed the case when replying normally as HTML:

![image](https://user-images.githubusercontent.com/52654513/74683000-e88b3380-51c7-11ea-9f95-0440ad106be0.png)

But as plain-text reply, this looks like this:

![image](https://user-images.githubusercontent.com/52654513/74683055-0e183d00-51c8-11ea-94ff-72b7454b2726.png)

So yes, there is still a full line between "Best Regards" and "Chris". And an extra line after "Chris".  And an extra line between the cursor and "Best Regards".

**Comment by ckeydel** [comment]
Oh, and an extra line before the cursor as well.

**Comment by RealRaven2000** [comment]
to get a better idea of what is _really_ happening, I highly recommend installing the Add-on [ThunderHtmlEdit](https://addons.thunderbird.net/en-US/thunderbird/addon/thunderhtmledit/), where you can switch back and forth between the view and the HTML behind that.

Now with the HTML editor you an actually influence the distance for the `<div>` by using some css css rules such as margin-bottom and they only affect the HTML result but not the plain text one. Plain text is always working in terms of "whole lines" and there is nothing you can do about it. Bt you can avoid the extra line break  for instance by removing the  `&nbsp;` (non breaking space) after the `<br>` or using less `<br>` elements. You can then add additional vertical space to the bottom of the div with margin-bottom.

Example:

```
<style>
#ck { 
  margin-bottom: 12px;
}
</style>
 <div id="ck">
   %cursor%<br>
   %sig(html)%
 </div>
```

Now it obviously depends on the content of your sig on what's at the bottom and may generate extra spacing towards your quote header. You will only know when looking into the "generated HTML" in ThunderHTMLEdit (within Thunderbird Composer). You can check the actual values using the "inspector" tool of the main Thunderbird window (tools menu).

![image](https://user-images.githubusercontent.com/6314674/74684307-7d3f6280-51c3-11ea-8310-a832c69477d5.png)

You can then select the composer document from the document chooser on the top right:

![image](https://user-images.githubusercontent.com/6314674/74684496-fe96f500-51c3-11ea-80cd-5ca0a6810bea.png)

you can then use the "inspector" top left for clicking on elements in the composer window and the "Layout" tab on the right to tell us about what it does distance-wise:

![image](https://user-images.githubusercontent.com/6314674/74684580-3605a180-51c4-11ea-83e6-a12845c9cc3f.png)

the "box model" tells us the full truth:
![image](https://user-images.githubusercontent.com/6314674/74684656-5c2b4180-51c4-11ea-94d2-a88ff848a695.png)

as you see, this stuff isn't exactly easy or easy to explain, but it's well worth the learning curve...

**Comment by ckeydel** [comment]
I remember this working really well, waaay back with old versions of TB and the predecessor of your add-on. And then, little by little, as TB "progressed", it all fell apart.

Anyway, thanks for your explanations and I will look into it, but my main problem seems to be that I can't find a way to have HTML signatures where individual lines don't get separated by extra lines in plain text emails. Do you have an idea for this? Just multiple lines in HTML that stay that way in plain text.

**Comment by RealRaven2000** [comment]
maybe we have the same problem as in your quote header (I had to do some modifications to deal with that)?

Can you put your signature here (just mask personal info  such as email addresses with ***)

You can easily post multi-line code in github and then **highlight** and **press the insert code button** afterwards, which will wrap it with "triple apostrophes" which is markdown for "leave this alone" ;)

![image](https://user-images.githubusercontent.com/6314674/74686625-ffcb2080-51c9-11ea-9997-2217a86544c1.png)

(check the preview tab to try it out). Example for a sig (css rules stripped for brevity):

```
<div id="mySignature">
  <b class="myName">%identity(name,link)%</b>
  <br/>Thunderbird Daily
  <br/>
  <span class="AddonList">(<a href="https://addons.thunderbird.net/thunderbird/addon/quickfolders-tabbed-folders/">QuickFolders</a>, 
  <a href="https://addons.thunderbird.net/thunderbird/addon/quickfilters/">quickFilters</a>, 
  <a href="https://addons.mozilla.org/firefox/addon/quickpasswords/">QuickPasswords</a>, 
  <a href="https://addons.thunderbird.net/thunderbird/addon/zombie-keys/">Zombie Keys</a>, 
  <a href="https://addons.thunderbird.net/thunderbird/addon/smarttemplate4/">SmartTemplate⁴</a>)</span>
  <br/>
  Visit my <a href="https://www.youtube.com/c/thunderbirddaily">YouTube Channel</a> for email productivity tips

  <img style="margin-top: 1em; float: right; box-shadow: 1px 1px 2px rgba(20, 20, 20, 0.4);" 
      moz-do-not-send="false"
      class="myLogo"  
      src="file:///C:/Users/Axel/Documents/Thunderbird/Signatures/thunderbird_blog2.png" 
      alt="Get Thunderbird!" 
      height="15" 
      width="94">
</div>
```

that one gets nicely "stripped down" in plain text mode here (I probably have to set up my test identity better so that %identity(name)% is properly resolved, but it may be another bug i need to investigate:

```
-- 
*%identity(name,link)%*
Thunderbird Daily
(QuickFolders
<https://addons.thunderbird.net/thunderbird/addon/quickfolders-tabbed-folders/>,
quickFilters
<https://addons.thunderbird.net/thunderbird/addon/quickfilters/>,
QuickPasswords
<https://addons.mozilla.org/firefox/addon/quickpasswords/>, Zombie Keys
<https://addons.thunderbird.net/thunderbird/addon/zombie-keys/>,
SmartTemplate⁴
<https://addons.thunderbird.net/thunderbird/addon/smarttemplate4/>)
Visit my YouTube Channel <https://www.youtube.com/c/thunderbirddaily>
for email productivity tips Get Thunderbird!

```

**Comment by RealRaven2000** [comment]
Ok I spoke too soon - I actually tested while I had the template disabled on the accounbt, I am getting a very interesting "bastardized" mixture in plain text mode. First in HTML , for comparison:

![image](https://user-images.githubusercontent.com/6314674/74687299-ed51e680-51cb-11ea-8eed-88ec4122783e.png)

and now in "plain text" (ish) mode - holding down SHIFT while replying:

![image](https://user-images.githubusercontent.com/6314674/74687343-0c507880-51cc-11ea-94ab-b5cb56310061.png)

So there still are plenty of extra lines in the signature box and even the body of the template itself,. Looks like I only solved the problem for the quote header. but that's to be expected when we fix a single thing only. I will keep testing. Still don't quite understand why there is even styling shown in a "plain text" window... 

plain text, the unloved stepchild of SmartTemplate4 ;)

**Comment by RealRaven2000** [comment]
OK, here is a trial version:

[smartTemplate-fx-2.9.2pre15.zip](https://github.com/RealRaven2000/SmartTemplate4/files/4216549/smartTemplate-fx-2.9.2pre15.zip)

... which is hopefully a little bit better in plain text. (not quite there yet, but closer)

improved output in plain text mode, with definitely a lot less uselss line breaks:

![image](https://user-images.githubusercontent.com/6314674/74689759-46257d00-51d4-11ea-9a6c-d7c1394fc2f0.png)

**Comment by ckeydel** [comment]
Yes, this version fixes it for me - great!

**Comment by RealRaven2000** [comment]
> Why do you say "not quite there yet"? AFAICS this version handles line breaks perfectly - I am getting exactly the same in both HTML and plain-text responses.

If you look at the screenshot of the signature there is a line break after the "Quickpasswords" Hyperlink. AFAIK it should break after 72 characters, so this looks a little early to me. Also the leading spaces are definitely ugly. I wonder if HTML should at least "collapse" all white spaces to a single one. Finally replacing newlines with nothing my lead to stuff "caking together" so replacement with a single space may be better. My safeguard at the moment is (not ideal) that I only do these replacements if the HTML snippet contains `<br>` or `<p>` tags (which may be overly cautious).

> BTW the fact that one is getting HTML-fomatted text for the reply header, signature etc. in a plain-text compose window is one of the oddities that TB introduced at some point. I remember "back in the old days", a plain-text compose window really meant plain-text when writing your email. But honestly, as long as I can rely on the format being ok when the email goes out, I don't care anymore. Just one more strange thing to live with I guess.

it would be nice to have an alternative (text-only) alternative signature for this case but that also means additional identity management work for the user. The whole point about using CSS rules for HTML is that it "degrades gracefully" so that it still looks meaningful and well layouted when a mail client can only display text-only mails.

**Comment by ckeydel** [comment]
>Also the leading spaces are definitely ugly.

Pretty sure these come from the indentation in your HTML template. I have manually removed them in mine - the HTML works just as well without. But yes, it would be nice if they could stay in without affecting the plain-text version.
> it would be nice to have an alternative (text-only) alternative signature for this case

True

**Comment by RealRaven2000** [comment]
> > Also the leading spaces are definitely ugly.
> 
> Pretty sure these come from the indentation in your HTML template. I have manually removed them in mine - the HTML works just as well without. But yes, it would be nice if they could stay in without affecting the plain-text version.

Would it be fair to say these are only leading spaces following block level elements / elements that insert line breaks? There may be a regular expression to take care of that...

**Comment by RealRaven2000** [comment]
So I found a regular expression for replacing the leading spaces (and tabs) too; here is the new replacement routine:

```
function cleanPlainTextNewLines(myHtml) {
  let lc = myHtml.toLocaleLowerCase();
  if (lc.includes("<br") || lc.includes("<p"))
    return myHtml.replace(/(\r\n)+|\r+|\n+|^[ \t]+/gm,""); 
  return myHtml;
}
```

Leading to a cleaner output (from the same HTML signature):
![image](https://user-images.githubusercontent.com/6314674/74776654-de346c80-528f-11ea-8ded-e73a32bfe62e.png)


To force leading spaces, you can still use the `&nbsp;` code. Tryout version:
[smartTemplate-fx-2.9.2pre16.zip](https://github.com/RealRaven2000/SmartTemplate4/files/4221607/smartTemplate-fx-2.9.2pre16.zip)

**Comment by ckeydel** [comment]
Yes, good so far. Just noticed there are extra spaces in the plain-text reply header where there shouldn't be. For example, this is the HTML version:

![image](https://user-images.githubusercontent.com/52654513/74914307-821e3500-53c2-11ea-9d69-98e44dee4cfd.png)

And here is the plain-text version of the same replied-to email:

![image](https://user-images.githubusercontent.com/52654513/74914375-a1b55d80-53c2-11ea-84de-10410d9e8499.png)

I marked the extra space in yellow. Why the extra space only in the "Sent:" line? Well, I am using this construct:

`<b>Sent:</b> %X:=sent% %A%, %d%.%m%.%Y% %H%:%M% UTC<br>`

Yes, there is a space before and after `%X:=sent%` and if I remove either one, it works as expected in both cases, e.g.:

`<b>Sent:</b>%X:=sent% %A%, %d%.%m%.%Y% %H%:%M% UTC<br>`

But is is different behaviour between HTML and plain-text, nonetheless, and may also happen with other constructs that I am not using.

**Comment by RealRaven2000** [comment]
> I marked the extra space in yellow. Why the extra space only in the "Sent:" line? 

Because HTML "collapses" all spaces (and line breaks and tabs) into one single space. I was debating with myself whether I should do this during the plain text conversion or not; obviously getting rid of leading space was already a good start. But should all spaces be collapsed?


> Well, I am using this construct:
> 
> `<b>Sent:</b> %X:=sent% %A%, %d%.%m%.%Y% %H%:%M% UTC<br>`
> 
> Yes, there is a space before and after `%X:=sent%` and if I remove either one, it works as expected in both cases, e.g.:

Also there may be an additional difficulty because the plain text conversion happens _before_ SmartTemplate replaces its variables (IIRC) so it won't know that there will be a double space. Notwithstanding of me fixing the space issue I would recommend this more comfortable way of writing that:

`%X:=sent%<b>Sent:</b>  %dateformat(A, d.m.Y H:M)% UTC<br>
`

In v 2.8, I have provided some multi-lingual examples on the variables tab, under the caption "_Variables for customized 'Date format'_":

![image](https://user-images.githubusercontent.com/6314674/74925256-c95cf380-53cb-11ea-819e-fedda89f0d6a.png)

you can just click them to insert into your template.

**Comment by ckeydel** [comment]
Yes, using %dateformat% is much nicer, thanks!

You're probably right - some people may want to use extra spacing.

**Comment by ckeydel** [comment]
OT: converter.html2txt.structs=false removes the TB-specific plain-text formatting such as using enclosing '*' for bold text:
![image](https://user-images.githubusercontent.com/52654513/75017904-45b80b00-548e-11ea-8f32-81f143b4ac70.png)
so that it looks ... well, plain:
![image](https://user-images.githubusercontent.com/52654513/75017993-70a25f00-548e-11ea-8095-15926d6ad66b.png)
... just adding this here for those plain-text aficionados that may stumble across this.

**Comment by RealRaven2000** [comment]
Fixed in 2.10.1 released on 14/04/2020
